handsontable/handsontable
# Fix canonicals and auto-redirects

### Context

The PR takes another step in unifying the Docs building process between _watch_ and _build_ modes.

#### Before
Docs in _watch_ mode:
 * Generated from `./docs/.watch-tmp/**/*.md`;
 * The content generation is built once for all frameworks. Thanks to that, the plugins such as `generate-nginx-redirects` and `dump-docs-data` see all pages and can produce valid results;

Docs in _build_ mode:
 * Generated from `./docs/content/**/*.md`;
 * The content generation is built for every framework in separate processes. The approach brakes the `generate-nginx-redirects` and `dump-docs-data` plugins as they do not have information about all pages (for all frameworks) in this specific Docs version;

The approach uses many `if`'s in the code that checks the current mode. That increases the maintenance cost and decreases the code readability. In this approach, the _watch_ mode goes on its own path and the _build_ mode on another.

#### After
Docs in _watch_ and _build_ mode work the same:
 * Generated from `./docs/.build-tmp/**/*.md`;
 * The content generation is built once for all frameworks. Thanks to that, the plugins such as `generate-nginx-redirects` and `dump-docs-data` see all pages and can produce valid results.
 * There is no usage for env `DOCS_FRAMEWORK`;
 * It's easier to add a new plugin or change the code in the future as the modes work the same in both environments.

_[skip changelog]_

### How has this been tested?

I tested the changes using the staging environment. I checked the following things:
 * Check if the only framework-specific content is searchable;
 * Check if the sidebar contains only the framework-specific content;
 * Check canonical URLs (tested by generating the content for 12.1);
 * Check Demos and assets URLs;
 * Check how the content is generated in _watch_ and the _build_ modes (should be the same);

### Types of changes


### Related issue(s):
1. fixes #9756

### Checklist:


- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
